sqlmesh/core/model/definition.py

```diff
@@ -1880,21 +1880,20 @@ def _split_sql_model_statements(
     return query, expressions[:pos], expressions[pos + 1 :]
 
 
-def _resolve_custom_session_properties(
-    defaults: t.Optional[t.Dict[str, t.Any]],
-    provided: t.Optional[exp.Expression] | t.Optional[t.Dict[str, t.Any]],
+def _resolve_session_properties(
+    default: t.Optional[t.Dict[str, t.Any]],
+    provided: t.Optional[exp.Expression | t.Dict[str, t.Any]],
 ) -> t.Optional[exp.Expression]:
     if isinstance(provided, dict):
         session_properties = {k: exp.Literal.string(k).eq(v) for k, v in provided.items()}
+    elif provided:
+        session_properties = {expr.this.name: expr for expr in provided}
```

Comment:
what is expr? what are you expecting it to be such that expr.this.name works

Reply:
It's a tuple of properties.

Reply:
so we expect this to be `EQ` with the left operand being a string literal or an identifer

Reply:
sg